sirepo/package_data/static/js/sirepo-components.js

```diff
@@ -49,6 +49,12 @@ SIREPO.app.directive('advancedEditorPane', function(appState, panelState) {
             var viewInfo = appState.viewInfo($scope.viewName);
```

Comment:
Is there a way to avoid using $scope here? It seems like all these values could be on an object, and then refer to them via that object.

Reply:
You can think of $scope as self here. The args come in from html attributes, so you could grab "data-view-name" attribute from the html element. Attributes like modelData are somehow replaced with an actual object by angularjs, so that would be tricky to replace.

Reply:
The question is: do they have to come from html elements. Would it not be better to parameterize another way? I need to think about this.